### DISCUSSION TITLE
Automatic fallback to lower package version if no installation candidate can be found

### DISCUSSION THREAD

**Discussion by PhilippWillms** [python-poetry/poetry#7638] [Q&A]
Hi Poetry enthusiast,
today I encountered a situation which triggered some structural discussion regarding the poetry logic. The examplle below occurred with the `tensorflow `library, but you may find also other instances.

**Original situation**
During my own development process, I learnt that I need a matching combination of following packages:
- tensorflow
- tensorflow-intel
Further restriction is to have tensorflow < 2.11, as since then the Windows-native GPU support is no longer available.

**Challneges in toml file**
My original idea was of course to have a toml config like the one below. 

```
tensorflow = ">2.10,<2.11"
tensorflow_intel = {version = ">=2.10,<2.11", markers = "sys_platform == 'win32'"}
```
Note that I want to overcome all the security vulneravilities in tf 2.10.0. It is now the case that after a `poetry lock` execution, the `poetry install` sucessfully finds the tf 2.10.1 package, but also wants to install tensorflow_intel 2.10.1 - which is not existing.

>  Unable to find installation candidates for tensorflow-intel (2.10.1)

**My question & idea**
What do you think on a poetry mechanism that uses 2.10.0 as a fallback, i.e. the latest possible version available at the searched indexes matching the version constraints in the toml file?

I would like to avoid explicit single version restrictions in such cases.

**Comment by dimbleby** [comment]
you've misunderstood something, poetry's behaviour already is to select the most recent compatible version of packages.

Note that tensorflow-intel 2.10.1 does in fact exist - https://pypi.org/project/tensorflow-intel/2.10.1/#files - though I notice that it looks as though they forgot the python 3.9 distribution, perhaps that's your real problem

**Reply by PhilippWillms** [reply]
Thanks for the quick reply. I addressed this issue in the tensorflow group :)
